### PR TITLE
Branch Sort Meeting

### PR DESCRIPTION
**Pull request by jeremykhoo-NUS**
## Add Sort Meeting Functionality

This pull request adds the ability to sort meetings in the address book application by different attributes: datetime, location, description, and title. Users can specify which attribute to sort by using command line prefixes, and they can also choose to sort in reverse order by adding the letter "r" after the prefix.

- Added a new command called SortMeetingCommand that takes in a prefix to determine which attribute to sort by and a boolean to determine if the sorting should be done in reverse order.
- Created comparator objects for each attribute that can be used to sort the meetings in the filtered meeting list.
- Implemented a switch case statement in the execute method of the SortMeetingCommand class to determine which comparator to use based on the prefix specified by the user.
- Updated the `MESSAGE_USAGE` and `MESSAGE_SUCCESS` strings to provide instructions on how to use the new command and display a message indicating which attribute the meetings were sorted by.
- This pull request adds a new function `getFirstKey()` to the `ArgumentMultiMap` class and a new parser `SortMeetingParser` to support sorting of meetings.
`getFirstKey()` returns the first key in the `ArgumentMultiMap` object. This function is useful when there is only one argument expected and it is unknown which prefix it belongs to.

`SortMeetingParser` is a new parser that parses user input for the `sort` command with options for sorting meetings by title, date/time, location, or description. It returns a `SortMeetingCommand` object that contains the sorting prefix and whether to sort in reverse order.

The `SortMeetingParser` class includes a private helper function `arePrefixesValidUnique()` to ensure that only one sorting prefix is specified. It also includes an implementation of the `getAutocompleteSuggestion()` method from the `Parser` interface, which returns a `null` suggestion and `false` flag since autocomplete is not currently supported for this command.

These changes are needed to add sorting functionality to the application, which can improve usability for users with large numbers of meetings.

Please review and let me know if any changes are needed. Thank you!

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/SortMeetingParser.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            return false;
+        } else {
+            return Stream.of(prefixes).anyMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+        }
+    }
+
+    @Override
+    public AutocompleteResult getAutocompleteSuggestion(String args) {
+        return new AutocompleteResult(null, false);
+    }
```
Comment (Beebeeoii) [inline src/main/java/seedu/address/logic/parser/SortMeetingParser.java:35]: How about having autocompletion that allows the user to `TAB` and cycle through the available `Prefix` to sort by? something like this: 
```suggestion
@Override
    public AutocompleteResult getAutocompleteSuggestion(String args) {
        String[] argsSplit = args.trim().split(" ");
        if (args.isEmpty()) {
            return new AutocompleteResult(PREFIXES[0], false);
        }
        String lastPrefix = argsSplit[argsSplit.length - 1];

        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(" " + lastPrefix, PREFIXES);
        boolean returnNext = false;

        for (Prefix p : PREFIXES) {
            if (returnNext) {
                return new AutocompleteResult(p, true);
            }

            if (argMultimap.getValue(p).isPresent() && p.toString().equals(lastPrefix)) {
                returnNext = true;
            }
        }

        if (!args.endsWith("/")) {
            return new AutocompleteResult(null, false);
        }

        return new AutocompleteResult(PREFIXES[0], true);
    }
```

File: src/main/java/seedu/address/model/AddressBook.java [hunk 2]
```diff
@@ -132,6 +133,14 @@ public void addMeeting(Meeting m) {
         meetings.add(m);
     }
 
+    /**
+     * Sort Meeting in addressbook.
+     * The meeting must not already exist in the address book.
```
Comment (Beebeeoii) [inline src/main/java/seedu/address/model/AddressBook.java:138]: ooo what does this mean haha

File: src/main/java/seedu/address/logic/parser/ArgumentMultimap.java [hunk 3]
```diff
@@ -57,4 +58,19 @@ public List<String> getAllValues(Prefix prefix) {
     public String getPreamble() {
         return getValue(new Prefix("")).orElse("");
     }
+    /**
+    * Returns the number of entries in this ArgumentMultimap.
+    */
```
Comment (Beebeeoii) [inline src/main/java/seedu/address/logic/parser/ArgumentMultimap.java:63]:
```suggestion
    /**
     * Returns the number of entries in this ArgumentMultimap.
     */
```

File: src/main/java/seedu/address/logic/parser/ArgumentMultimap.java [hunk 4]
```diff
@@ -57,4 +58,19 @@ public List<String> getAllValues(Prefix prefix) {
...
+    public int getLength() {
+        return argMultimap.size();
+    }
+
+    /**
+    * Returns the number of entries in this ArgumentMultimap.
+    */
```
Comment (Beebeeoii) [inline src/main/java/seedu/address/logic/parser/ArgumentMultimap.java:70]:
```suggestion
    /**
     * Returns the number of entries in this ArgumentMultimap.
     */
```

**Review by Beebeeoii — CHANGES_REQUESTED** [review]
thanks for the PR! looks great. some a thought: sorting is default to descending order? should it be defaulted to ascending instead?

**Review by Beebeeoii — APPROVED** [review]
lgtm !

**Review by chuakid — APPROVED** [review]
LGTM

**Pull request merged**